contracts/utils/Strings.sol

```diff
@@ -323,7 +323,7 @@ library Strings {
     }
 
     /**
-     * @dev Variant of {tryParseHexUint} that does not check bounds and returns (true, 0) if they are invalid.
+     * @dev Variant of {tryParseHexUint} that does not check bounds and returns (true, 0) if `begin <= end` excluding `0x`.
```

Comment:
I don't understand.

I most cases (like the normal cases) we have "begin <= end" ... yet we don't necessarily return `(true, 0)` ... we return `(true, result)` with the result possibly being non zero.

It feels to me like the more we try to address audit comments, the more confused we get. I used to be confident in the code, that is no longer the case. IMO we should go back and have a look at what the code really does, instead of adding patches over patches.

In this particular what do we have:
- a function that tryes to parse uints in hex format
- it do not check the bounds
- if the input within bonds and contain a valid hex string, it returns (true, result)
- if the input within bonds and contains an invalid hex string, it returns (false, 0)
- if begin > end, it returns (true, 0)
- if end > input.length, it reads possibly unalocated memory.

Note that this is a private function, and that any documentation will NOT show up in docs.openzeppelin.com. So IMO we should either:
- not document at all (that used to be the rule for private function)
- be very factual and list the cases one by one as I did here.

Reply:
```suggestion
     * @dev Implementation of {tryParseHexUint} that does not check bounds. Caller should check that 
     * `begin <= end <= input.length`. Other inputs would result in unexpected behavior.
```